compiler/rustc_interface/src/queries.rs

```diff
@@ -240,7 +234,7 @@ impl<'tcx> Queries<'tcx> {
                 } else {
                     None
                 },
-                ongoing_codegen,
+                ongoing_codegen: passes::start_codegen(&*self.compiler.codegen_backend, tcx),
```

Comment:
Does the `tcx.dep_graph.clone()` produce an independent dep graph or just another reference to the main dep graph? If the former, this call need to be moved before the clone.

Reply:
`DepGraph` contains two fields that are both `Lrc`, so I think it's the latter. But I have moved the `start_codegen` call earlier just in case.